simplelru/lru.go

```diff
@@ -12,31 +12,52 @@ import (
 // EvictCallback is used to get a callback when a cache entry is evicted
 type EvictCallback[K comparable, V any] func(key K, value V)
 
-// LRU implements a non-thread safe fixed size LRU cache
-type LRU[K comparable, V any] struct {
+// Cache implements a non-thread safe fixed cache with Cache and SIEVE eviction (https://cachemon.github.io/SIEVE-website/)
+type Cache[K comparable, V any] struct {
 	size      int
 	evictList *internal.LruList[K, V]
 	items     map[K]*internal.Entry[K, V]
 	onEvict   EvictCallback[K, V]
+	hand      *internal.Entry[K, V]
+	useSieve  bool
 }
 
 // NewLRU constructs an LRU of the given size
-func NewLRU[K comparable, V any](size int, onEvict EvictCallback[K, V]) (*LRU[K, V], error) {
+func NewLRU[K comparable, V any](size int, onEvict EvictCallback[K, V]) (*Cache[K, V], error) {
 	if size <= 0 {
 		return nil, errors.New("must provide a positive size")
 	}
 
-	c := &LRU[K, V]{
+	c := &Cache[K, V]{
 		size:      size,
 		evictList: internal.NewList[K, V](),
 		items:     make(map[K]*internal.Entry[K, V]),
 		onEvict:   onEvict,
 	}
+
+	return c, nil
+}
+
+// NewSieve constructs a SIEVE of the given size
+func NewSieve[K comparable, V any](size int, onEvict EvictCallback[K, V]) (*Cache[K, V], error) {
```

Comment:
If you go ahead and rename to Cache anyways, think about having a single constructor with options.  That will clean up your documentation by reducing some clutter.  WithSieve() could be the option to turn this on.